frontend/ui/uimanager.lua

```diff
@@ -905,21 +911,24 @@ function UIManager:_checkTasks()
     -- Tasks due for execution might themselves schedule more tasks (that might also be immediately due for execution ;)).
     -- Flipping this switch ensures we'll consume all such tasks *before* yielding to input polling.
     self._task_queue_dirty = false
-    while self._task_queue[1] do
-        local task_time = self._task_queue[1].time
+    local task = self._task_queue[self._task_queue.len]
+    while task do
```

Comment:
`while self._task_queue.len > 0`?

(Mainly to avoid checking `self._task_queue[0]`, because that would break the "this is an array" contract ;).